doc/extension.rst

```diff
@@ -67,7 +68,11 @@ See also additional configuration options in the :ref:`built-in extensions
 
    A list of arguments to pass to ``clang`` while parsing the source, typically
    to add directories to include file search path, or to define macros for
-   conditional compilation. No arguments are passed by default.
+   conditional compilation.
+
+   No arguments are passed by default when processing C domain
+   directives. ``-xc++`` is passed as the first argument when processing C++
+   domain directives.
```

Comment:
Not sure about this paragraph. It's technically accurate, but strictly from the view of a user, I worry it raises more questions than not.

Reply:
I left this change out.